File: src/main/java/no/novari/flyt/archive/gateway/dispatch/web/FintArchiveDispatchClientConfigurationProperties.java
package no.novari.flyt.archive.gateway.dispatch.web;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;

@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ConfigurationProperties("novari.flyt.archive.gateway.dispatch.fint-client")
public class FintArchiveDispatchClientConfigurationProperties {
    private Duration postFileTimeout;
    private Duration postCaseTimeout;
    private Duration postRecordTimeout;
    private Duration getStatusTimeout;
    private Duration createdLocationPollTotalTimeout;
    private Duration createdLocationPollBackoffMinDelay;
    private Duration createdLocationPollBackoffMaxDelay;
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/dispatch/web/FintArchiveDispatchClientConfigurationProperties.kt
package no.novari.flyt.archive.gateway.dispatch.web

import org.springframework.boot.context.properties.ConfigurationProperties
import java.time.Duration

@ConfigurationProperties("novari.flyt.archive.gateway.dispatch.fint-client")
class FintArchiveDispatchClientConfigurationProperties {
    var postFileTimeout: Duration? = null
    var postCaseTimeout: Duration? = null
    var postRecordTimeout: Duration? = null
    var getStatusTimeout: Duration? = null
    var createdLocationPollTotalTimeout: Duration? = null
    var createdLocationPollBackoffMinDelay: Duration? = null
    var createdLocationPollBackoffMaxDelay: Duration? = null
}
